### DISCUSSION TITLE
Memory paging with a register

### DISCUSSION THREAD

**Discussion by jpelletier** [wwarthen/RomWBW#374] [Q&A] [answered]
I have implemented my own Z80 computer with paged memory. A common segment above 8000, and 32k pages mapped at 0000-7FFF.
Pages are addressed with an 8-bit IO register (R/W) at address FF, giving a total of 256 * 32k = 8M paged memory. 

How do we support this in RomWBW?

**Comment by wwarthen** [comment]
I can easily add the code to support this.

- Which of the 256 pages are ROM?
- Do you have a name for your system?

Thanks,

Wayne

**Reply by jpelletier** [reply]
ROM pages  are 10-1F (512k) but I can remap it easily. I eventually plan to regroup all ROMs in pages 80-FF and RAMs from 00-7F.
I nicknamed my system the monsputer :)
I documented it on sourceforge:
https://sourceforge.net/projects/monsputer-bios/

**Comment by wwarthen** [comment, marked as answer]
Hi @jpelletier,

I have just checked in the changes to begin supporting Monsputer.  Commit e286a42 in the dev branch.  The abbreviated platform name is "MON".  If you want to use something different, let me know.

There is nothing enabled in the MON configuration at this point.  You will need to review the configuration files and enable/configure what is needed for Monsputer.

I added the code for the memory paging to hbios.asm, but you should review what I did.  It is very simple.  I'm sure you will have questions, so please just ask.

Thanks,

Wayne
